### PR TITLE
Refactor MISC_SPRITE to strong enum

### PR DESCRIPTION
**Pull request by ZehMatt**
I had to keep the type of ```type``` as its has a different representation based on the entity type.

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
This is pretty similar to the window code that @tupaschoal and myself were discussing. For the window code we created a union with the idea being that the field is eventually split in a future refactor. I feel we should be consistent with that.

**Comment by ZehMatt** [comment]
I can't forward declare nested types, I initially attempted that but its better to split the variable later.

File: src/openrct2/world/MoneyEffect.cpp [hunk 1]
```diff
@@ -43,7 +43,7 @@ void MoneyEffect::CreateAt(money32 value, const CoordsXYZ& effectPos, bool verti
     moneyEffect->sprite_height_positive = 30;
     moneyEffect->sprite_identifier = SpriteIdentifier::Misc;
     moneyEffect->MoveTo(effectPos);
-    moneyEffect->type = SPRITE_MISC_MONEY_EFFECT;
+    moneyEffect->type = EnumValue(MiscSpriteType::MoneyEffect);
```
Comment (Gymnasiast) [inline src/openrct2/world/MoneyEffect.cpp:46]: Can we declare `type` as a MiscSpriteType instead?

**Comment by Gymnasiast** [comment]
Since we renamed sprites to entities, `MiscEntityType` would be a more appropriate name.

File: src/openrct2/world/MoneyEffect.cpp (hunk 1 above)
Comment (ZehMatt) [reply src/openrct2/world/MoneyEffect.cpp:46]: Vehicle uses this as Vehicle::Type, thats what I meant by splitting up the variable, Vehicle should have its own type and Misc should have its own type.

File: src/openrct2/paint/sprite/Paint.Misc.cpp [hunk 2]
```diff
@@ -134,7 +134,9 @@ void misc_paint(paint_session* session, const SpriteBase* misc, int32_t imageDir
                 isAntiClockwise = !isAntiClockwise;
             }
 
-            uint32_t baseImageId = (jumpingFountain->type == SPRITE_MISC_JUMPING_FOUNTAIN_SNOW) ? 23037 : 22973;
+            uint32_t baseImageId = (static_cast<MiscEntityType>(jumpingFountain->type) == MiscEntityType::JumpingFountainSnow)
+                ? 23037
```
Comment (Gymnasiast) [inline src/openrct2/paint/sprite/Paint.Misc.cpp:138]: These naked numbers should get a constant. Like `SPR_JUMPING_FOUNTAIN_WATER_START` and `SPR_JUMPING_FOUNTAIN_SNOW_START`.

File: src/openrct2/world/MoneyEffect.cpp [hunk 3]
```diff
@@ -1,4 +1,4 @@
-/*****************************************************************************
+﻿/*****************************************************************************
```
Comment (Gymnasiast) [inline src/openrct2/world/MoneyEffect.cpp:1]: *cough*

File: src/openrct2/world/Sprite.cpp [hunk 4]
```diff
@@ -1,4 +1,4 @@
-/*****************************************************************************
+﻿/*****************************************************************************
```
Comment (Gymnasiast) [inline src/openrct2/world/Sprite.cpp:1]: Again.

Can you please stop your IDE from inserting these silly BOMs, which should not be present in UTF-8 files?

Comment (ZehMatt) [reply src/openrct2/world/Sprite.cpp:1]: I have no bloody idea how it keeps doing this, I always put it back but at some point it just screws me over again.

File: src/openrct2/paint/sprite/Paint.Misc.cpp (hunk 2 above)
Comment (ZehMatt) [reply src/openrct2/paint/sprite/Paint.Misc.cpp:138]: I'll do that in another PR, there are more naked numbers in this file like that.

**Comment by duncanspumpkin** [comment]
What about a getter function instead of the casts?

**Review by Gymnasiast — APPROVED** [review]

**Comment by ZehMatt** [comment]
> What about a getter function instead of the casts?

The base class has no idea about how the type member is treated.

**Pull request merged**